### DISCUSSION TITLE
The team have a extremely limited test capabilities in which COM is used

### DISCUSSION THREAD

**Discussion by kant2002** [dotnet/winforms#6809] [Q&A] [answered]
We (the team) have a extremely limited test capabilities and limited understanding of real scenarios in which COM is used in conjunction with Windows Forms apps. This makes it very difficult to accept changes in this area without good and verifiable test coverage.

_Originally posted by @RussKie in https://github.com/dotnet/winforms/issues/6743#issuecomment-1061349988_

**Comment by kant2002** [comment]
I thinking about other areas where I would like to have COM related tests. Based on my experience in dotnet/runtime repo where they have more interesting infrastructure for testing managed/unmanaged interactions.

I would like to discuss idea how that part of test infra can be improved. My preference would be to have separate projects for set of test cases (or control) where similar COM object implementation can live. Having one project for native side would be unsustainable.

I plan at least try to make it testable in automated fashion, but https://github.com/dotnet/winforms/blob/main/src/System.Windows.Forms/tests/IntegrationTests/NativeHost/readme.md indicates that this can be problematic. Any thoughts on this?

Did you consider using CMake for building native projects?

**Comment by RussKie** [comment, marked as answer]
We're using CMake in src\System.Windows.Forms\tests\InteropTests\System.Windows.Forms.Interop.Tests.csproj (added by @hughbe in https://github.com/dotnet/winforms/pull/3732). If you able to convert https://github.com/dotnet/winforms/blob/main/src/System.Windows.Forms/tests/IntegrationTests/NativeHost to CMake - please do. I don't remember why it wasn't done as one.
In general, we will gladly accept any help with COM or native tests we can get :)

Thank you.
